tests/test_core.py

```diff
@@ -36,64 +38,111 @@ def test_inspect_dummy_execs():
 
 
 @pytest.mark.parametrize(
-    'package_path, expected_path',
+    'package_path, expected_path, build_args_envs, dry_run',
     [
         (
             Path(__file__).parent / 'cases' / 'executor_1',
             Path(__file__).parent / 'cases' / 'executor_1.json',
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            } ,
+            False
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_2',
             Path(__file__).parent / 'cases' / 'executor_2.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_3',
             Path(__file__).parent / 'cases' / 'executor_3.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_4',
             Path(__file__).parent / 'cases' / 'executor_4.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_5',
             None,
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'executor_6',
             None,
+            {},
+            True
+        ),
+        (
+            Path(__file__).parent / 'cases' / 'executor_7',
+            None,
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            },
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested',
             Path(__file__).parent / 'cases' / 'nested.json',
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_2',
             None,
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            },
+            False
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_3',
             None,
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_4',
             None,
+            {},
+            True
         ),
         (
             Path(__file__).parent / 'cases' / 'nested_5',
             None,
+            {},
+            True
         ),
     ],
 )
-def test_get_executor_args(package_path, expected_path):
+def test_get_executor_args(package_path, expected_path, build_args_envs, dry_run):
```

Comment:
I think we should create a new test function to test this instead of adding more and more parameters to this one.